app/services/action_builder.rb

```diff
@@ -72,11 +73,8 @@ def is_recurring_donation?
 
   def form_data
     @params.tap do |params|
-      member = nil
-      member = Member.find_by_akid(params[:referring_akid]) if params[:referring_akid].present?
-      member = Member.find_by(id: params[:rid]) if params[:rid].present?
-      member = Member.find_by(id: params[:referrer_id]) if params[:referrer_id].present?
-      params[:action_referrer_email] = member.email if member.try(:email).present?
+      email = MemberEmailGuesser.run(params)
```

Comment:
this is great. my big anxiety about what you said you were doing with changing the handling of params between the processor and the main app was that I was worried this code would get replicated in multiple places. Instead, you put it in a tidy little class. Big thumbs up from me. 👍  I'm looking forward to when we get that class moved over to the processor.